samples/nrf9160/pdn/src/main.c

```diff
@@ -25,6 +29,65 @@ static const char * const event_str[] = {
 	[PDN_EVENT_IPV6_DOWN] = "IPv6 down",
 };
 
+static void ifaddrs_print(void)
+{
+	int err;
+
+	struct nrf_ifaddrs *ifaddrs;
+	struct nrf_ifaddrs *ifa;
+
+	char addr[NRF_INET6_ADDRSTRLEN] = {0};
+	char netmask[NRF_INET6_ADDRSTRLEN] = {0};
+	char broadaddr[NRF_INET6_ADDRSTRLEN] = {0};
+	char dstaddr[NRF_INET6_ADDRSTRLEN] = {0};
+
+	err = nrf_getifaddrs(&ifaddrs);
+	if (err) {
+		printk("nrf_getifaddrs() failed, err %d\n", err);
+		return;
+	}
+
+	printk("\nInterface addresses:\n");
+
+	ifa = ifaddrs;
+	while (ifa != NULL) {
+		switch (ifa->ifa_netmask->sa_family) {
```

Comment:
I worked out why the output looks wrong ... this switch should be ifa->ifa_addr->sa_family.

Netmask is not set, so has default family (IPv), so the first block is being running, passing in the correct IPv6 family (so it looks like an IPv6 format), but passing in &SIN(ifa->ifa_addr)->sin_addr, which is why the bytes are wrong.

Reply:
This is the output, with the correct selector (and also outputting the family in the print):

These are correct looking IPv6 addresses.

```
Connected via Serial Port with settings /dev/ttyACM0 115200 8n1 rtscts:off

Event: PDP context 0 activated
Default APN is telstra.internet
Created new PDP context 1
PDP context 1 configured: APN telstra.internet, Family IPV4V6
Event: PDP context 1 activated
Event: PDP context 0 IPv6 up
Event: PDP context 1 IPv6 up
PDP Context 0, PDN ID 0
PDP Context 1, PDN ID 0

Interface addresses:
Address (1): 10.212.179.190
Netmask (1): 0.0.0.0
Broadcast address: 0.0.0.0
Destination address: 0.0.0.0

Address (2): fe80::f403:18b9:108f:34b
Netmask (1): 0.0.0.0
Broadcast address: 0.0.0.0
Destination address: 0.0.0.0

Address (2): 2001:8004:1180:3f28:f403:18b9:108f:34b
Netmask (1): 0.0.0.0
Broadcast address: 0.0.0.0
Destination address: 0.0.0.0

Event: PDP context 0 deactivated
Event: PDP context 1 deactivated
Bye
```

Reply:
Thanks, @sgryphon. I'll fix this now!